include/cantera/oneD/refine.h

```diff
@@ -69,51 +123,132 @@ class Domain1D;
         return m_gridmin;
     }
 
-    //! Determine locations in the grid that need additional grid points.
-    //!
-    //! @returns The number of new grid points needed (size of #m_loc)
+    /**
+     * Determine locations in the grid that need additional grid points and
+     * updates the internal state of the Refiner with this information.
+     *
+     * @param[in] n  Number of grid points
+     * @param[in] z  Point to array of grid points
+     * @param[in] x  Pointer to solution vector
+     *
+     * @returns The number of new grid points needed (size of #m_insertion_points)
+     */
     int analyze(size_t n, const double* z, const double* x);
+
+    /**
+     * Constructs a new grid based on refinement locations determined by the analyze
+     * method.
+     *
+     * This function generates a new grid by inserting additional points into the
+     * current grid at locations where the `analyze` method has identified a need for
+     * refinement. The new grid points are placed midway between existing points deemed
+     * necessary for increased resolution. If no refinement is needed, the original
+     * grid is copied directly.
+     *
+     * @param[in] n The number of points in the original grid array `z`.
+     * @param[in] z Pointer to the array of original grid points.
+     * @param[in] nn The maximum number of points that the new grid array `znew` can hold.
+     * @param[out] znew Pointer to the array where the new grid points will be stored.
+     *
+     * @return The function returns 0 upon successful creation of the new grid. Throws an exception if the provided
+     *         output array size is insufficient to hold the new grid.
+     */
     int getNewGrid(int n, const double* z, int nn, double* znew);
-    int nNewPoints() {
-        return static_cast<int>(m_loc.size());
+
+    /**
+     * Returns the number of new grid points that were needed.
+     */
+    int numNewPoints() {
```

Comment:
```suggestion
    int nNewPoints() {
```
Avoid unnecessary churn - we widely use the shorthand `n` to indicate number, so please retain the original version. Also, as this is part of the `public` interface, you would have to go through a deprecation cycle to avoid breakage of external user code.